src/util/customTransformer.js

```diff
@@ -285,12 +285,14 @@ async function userTransformHandler(
       events.forEach((ev) => {
         eventsMetadata[ev.message.messageId] = ev.metadata;
       });
+      const credentials = events[0].credentials;
```

Comment:
Why are we taking only from the first element?
Are credentials expected to be sent with every event? Why?

Reply:
All events are grouped by source and destination, so each group of events will have the same credentials. I still need to validate this.